## 初期設定

In [ ]:
import sys, os
from IPython.display import display, HTML, clear_output, update_display, Image
import numpy as np
import matplotlib.pyplot as plt

# 自作モジュール
sys.path.append(r"C:\Users\okaza\pythonenv")
from modules.Mytools.Tools import dict_tree, h5_tree
from modules.Mytools.handle_ipynb import save_pickle, load_pickle, export_html, ask_openfilename, ask_savefilename
sys.path.append(os.getcwd())

# 初期パラメーター
cachedir = os.path.join(os.getcwd(), ".cache")
os.makedirs(cachedir, exist_ok=True)

# データ格納用パラメーター
data = dict()

## 目的

PDIndexerのfittingした値を保存する

In [3]:
import pyperclip

In [15]:
import pandas as pd
data = dict()

In [16]:
def load():

    l = pyperclip.paste().split(sep = "\n")
    # print(l)
    s = pd.Series(l[1:])
    df = s.str.split(pat = "\t", expand = True)
    df.columns = l[0].split(sep = "\t")

    data["peaks"] = df.T.to_dict()
    dict_tree(data)
    

    return
load()
del load

<class 'dict'>  (size = 0.01 MB)
└── peaks
    ├── 0
    │   ├── Checked <class 'str'>
    │   ├── hkl <class 'str'>
    │   ├── Calculated X <class 'str'>
    │   ├── Function <class 'str'>
    │   ├── Observed X <class 'str'>
    │   ├── Observed X err (1sigma) <class 'str'>
    │   ├── Observed FWHM <class 'str'>
    │   ├── Observed Intensity <class 'str'>
 <class 'str'>p (%)
    ├── 1
    │   ├── Checked <class 'str'>
    │   ├── hkl <class 'str'>
    │   ├── Calculated X <class 'str'>
    │   ├── Function <class 'str'>
    │   ├── Observed X <class 'str'>
    │   ├── Observed X err (1sigma) <class 'str'>
    │   ├── Observed FWHM <class 'str'>
    │   ├── Observed Intensity <class 'str'>
 <class 'str'>p (%)
    ├── 2
    │   ├── Checked <class 'str'>
    │   ├── hkl <class 'str'>
    │   ├── Calculated X <class 'str'>
    │   ├── Function <class 'str'>
    │   ├── Observed X <class 'str'>
    │   ├── Observed X err (1sigma) <class 'str'>
    │   ├── Observed FWHM <class 'str'>
  

In [9]:
import json

In [17]:
print(json.dumps(data, indent = 4))

{
    "peaks": {
        "0": {
            "Checked": "True",
            "hkl": "0 1 0",
            "Calculated X": "12.090313356390633",
            "Function": "Sym PV",
            "Observed X": "12.08427563194857",
            "Observed X err (1sigma)": "0.2130198758566942",
            "Observed FWHM": "0.05206627425469563",
            "Observed Intensity": "789.1508591053409",
            "Rwp (%)\r": "0.0031519180812481984\r"
        },
        "1": {
            "Checked": "False",
            "hkl": "0 0 2",
            "Calculated X": "13.021754961047842",
            "Function": "Sym PV",
            "Observed X": "13.102513322981908",
            "Observed X err (1sigma)": "0.8484241297675529",
            "Observed FWHM": "0.1719038321252705",
            "Observed Intensity": "1297.2879995992855",
            "Rwp (%)\r": "0.006324619223690986\r"
        },
        "2": {
            "Checked": "True",
            "hkl": "0 1 1",
            "Calculated X": "13.738487

Checked	hkl	Calculated X	Function	Observed X	Observed X err (1sigma)	Observed FWHM	Observed Intensity	Rwp (%)
True	0 1 0	12.090313356390633	Sym PV	12.08427563194857	0.2130198758566942	0.05206627425469563	789.1508591053409	0.0031519180812481984
False	0 0 2	13.021754961047842	Sym PV	13.102513322981908	0.8484241297675529	0.1719038321252705	1297.2879995992855	0.006324619223690986
True	0 1 1	13.738487879922872	Sym PV	13.743419823736419	0.1085652259521828	0.060463183763789244	1912.8163549292024	0.004367939661382365
False	0 1 2	17.804900477474543	Simple	0	18.615280191266383	0.16493818174165387	32.65811866278883	0.0006072797436552028
True	1 1 0	21.0198866693487	Sym PV	21.020908801818344	0.9684274497893738	0.07790236166414446	296.8732114744767	0.0013532414654879102
False	0 1 3	23.0799155889439	Sym PV	0	0	0	0	0
False	0 2 0	24.317903135704697	Sym PV	0	0	0	0	0
False	1 1 2	24.80504155476072	Simple	0	0.006591797892196993	0.1	0	0
False	0 2 1	25.197478134623683	Simple	0	0.006591797892196993	0.1	0	0
False	0 0 4	26.215574138406925	Sym PV	0	0	0	0	0
False	0 2 2	27.67958111866387	Sym PV	0	0	0	0	0
False	0 1 4	28.960114405382416	Sym PV	0	0	0	0	0
